### PR TITLE
Add manage providers ans modules permissions

### PR DESCRIPTION
**Pull request by ministicraft**
## Description

Add manage providers and modules team permissions

## External links

- [go-tfe documentation](https://pkg.go.dev/github.com/hashicorp/go-tfe#OrganizationAccess)
- [API documentation](https://www.terraform.io/cloud-docs/users-teams-organizations/permissions#organization-permissions)
- [Related PR](https://github.com/hashicorp/go-tfe/pull/306)

### CODE REVIEW

**Comment by brandonc** [comment]
We'll need to update the go-tfe module dependency once your other change is released before this can be tested. Thanks again

File: website/docs/r/team.html.markdown [hunk 1]
```diff
@@ -48,6 +48,8 @@ The `organization_access` block supports:
 * `manage_policy_overrides` - (Optional) Allows members to override soft-mandatory policy checks.
 * `manage_workspaces` - (Optional) Allows members to create and administrate all workspaces within the organization.
 * `manage_vcs_settings` - (Optional) Allows members to manage the organization's VCS Providers and SSH keys.
+* `manage_providers` - (Optional) Allow members to publish and delete providers in the organization"s private registry.
```
Comment (brandonc) [inline website/docs/r/team.html.markdown:51]: " -> '
```suggestion
* `manage_providers` - (Optional) Allow members to publish and delete providers in the organization's private registry.
```

**Review by brandonc — COMMENTED** [review]
Can you also go ahead and update the go-tfe module to v0.25.0?

`go get -u github.com/hashicorp/go-tfe`

**Comment by ministicraft** [comment]
go-tfe release 0.25.0 seems to break something unrelated with this PR
`$ make test
==> Checking that code complies with gofmt requirements...
go test -v $(go list ./... |grep -v 'vendor') || exit 1
# github.com/hashicorp/terraform-provider-tfe/tfe
tfe/resource_tfe_ssh_key.go:90:3: unknown field 'Value' in struct literal of type tfe.SSHKeyUpdateOptions
# github.com/hashicorp/terraform-provider-tfe/tfe [github.com/hashicorp/terraform-provider-tfe/tfe.test]
tfe/resource_tfe_ssh_key.go:90:3: unknown field 'Value' in struct literal of type tfe.SSHKeyUpdateOptions
FAIL    github.com/hashicorp/terraform-provider-tfe/tfe [build failed]
GNUmakefile:12: recipe for target 'test' failed
make: *** [test] Error 1`

**Comment by brandonc** [comment]
@ministicraft yeah, there was a bug fix that broke a struct. I incorporated your commits into a new PR that I am updating. I can take it from here!

**Comment by brandonc** [comment]
Merged in #432

**Pull request closed without merging**